synapse/app/complement_fork_starter.py

```diff
@@ -167,13 +179,29 @@ def main() -> None:
     update_proc.join()
     print("===== PREPARED DATABASE =====", file=sys.stderr)
 
+    processes: List[multiprocessing.Process] = []
+
+    # Install signal handlers to propagate signals to all our children, so that they
+    # shut down cleanly. This also inhibits our own exit, but that's good: we want to
+    # wait until the children have exited.
```

Comment:
I don't see why our own exit is blocked waiting for our children to exit. Does `os.kill()` block until the child handles the signal? Or does we already have this for free thanks to `process.join()` further below?